addons/account/models/account_account.py

```diff
@@ -964,15 +939,46 @@ def _unlink_except_contains_journal_items(self):
     @api.ondelete(at_uninstall=False)
     def _unlink_except_account_set_on_customer(self):
         #Checking whether the account is set as a property to any Partner or not
-        values = ['account.account,%s' % (account_id,) for account_id in self.ids]
-        partner_prop_acc = self.env['ir.property'].sudo().search([('value_reference', 'in', values)], limit=1)
-        if partner_prop_acc:
-            account_name = partner_prop_acc.get_by_record().display_name
-            raise UserError(
-                _("You can't delete the account %s, as it is used on a contact.\n\n"
-                    "Think of it as safeguarding your customer's receivables; your CFO would appreciate it :)"
-                    , account_name)
-            )
+        for field in self.env.registry.many2one_company_dependents[self._name]:
```

Comment:
Shouldn't there be a helper for this? I'm surprised it's not needed at many places